### PR TITLE
Support clearing config

### LINKED ISSUE #68: Support clearing config
Overheard from @guybedford:

If explicit [auth / registry] configuration is present, configure wizard should have a prompt which allows the user to clear this option.

### PR DESCRIPTION
**Pull request by crisptrutski**
Closes #68

### CODE REVIEW

**Comment by crisptrutski** [comment]
@guybedford some noise from first commit, just look at the second one - this is exactly your proposal I believe

**Comment by crisptrutski** [comment]
@guybedford ping

File: npm.js [hunk 1]
```diff
@@ -172,25 +172,55 @@ NPMLocation.configure = function(config, ui) {
...
   // check if there are settings in npmrc
-  return rsvp.resolve(config.registry || npmrc.getRegistry() || defaultRegistry)
-  .then(function(resolvedRegistry) {
-    return ui.input('npm registry', resolvedRegistry);
-  })
-  .then(function(registry) {
-    config.registry = registry;
-    if (config.registry.substr(config.registry.length - 1, 1) == '/')
-      config.registry = config.registry.substr(0, config.registry.length - 1);
+  return rsvp.resolve()
```
Comment (guybedford) [inline npm.js]: This should probably just be `Promise.resolve()`

File: npm.js [hunk 2]
```diff
@@ -172,25 +172,55 @@ NPMLocation.configure = function(config, ui) {
...
-    if (!auth)
+  .then(function(useNpmrc) {
+    if (useNpmrc) {
+      delete config.registry;
+      delete config.auth;
       return;
+    }
 
-    return configureCredentials(config.registry, npmrc.getAuth(config.registry), ui)
+    return rsvp.resolve(config.registry || defaultRegistry)
```
Comment (guybedford) [inline npm.js]: Promise.resolve

**Comment by guybedford** [comment]
:+1:

File: npm.js (hunk 1 above)
Comment (crisptrutski) [inline npm.js]: Wow, jam packed namespaces and aliasocity! Didn't realise the tool was at hand, will drop the extra import.

**Pull request merged**